src/jvmMain/kotlin/com/charleskorn/kaml/YamlOutput.kt

```diff
@@ -96,8 +96,8 @@ internal class YamlOutput(
 
     override fun encodeEnum(enumDescriptor: SerialDescriptor, index: Int) = emitQuotedScalar(enumDescriptor.getElementName(index), configuration.singleLineStringStyle.scalarStyle)
 
-    private fun emitPlainScalar(value: String) = emitScalar(value, ScalarStyle.PLAIN)
-    private fun emitQuotedScalar(value: String, scalarStyle: ScalarStyle) = emitScalar(value, scalarStyle)
+    private fun emitPlainScalar(value: String) = emitScalar(value, ScalarStyle.PLAIN, configuration.yamlNamingStrategy)
+    private fun emitQuotedScalar(value: String, scalarStyle: ScalarStyle) = emitScalar(value, scalarStyle, configuration.yamlNamingStrategy)
```

Comment:
Should this only apply to field names? At present, we'll apply the naming strategy to all field names and also all scalar values.

Reply:
I made the mistake of applying this to the values, not the serial names. I need to look into this to see how to change the serial names. The algorithms to transform a string into a case type work well though so it's just finding a way to apply it.

Reply:
IMHO this should only be applied to field names

Reply:
Just to confirm, this will ignore the @SerializeName?

I'd like to be able to use the same data model classes for JSON and YAML serialization (I've done this with Jackson before).
Alternative is @SerializeName should be able to specify json/yaml variants

Reply:
I think it makes the most sense for this to be applied to all items, but `@SerialName` be an override to this

Reply:
`@SerialName` is resolved at compile time,there isn't a way to know if it was applied to a specific value or not, so strategies will ignore it. This is how it works with KotlinX Serialization's Json one too. Naming strategies should cover most use cases apart from when there is inconsistent naming.